### PR TITLE
Updated to v2024.0 public ovms package in C-API Minimal App Demo

### PR DESCRIPTION
**Pull request by bstrzele**
CVS-130902

### CODE REVIEW

**Review by atobiszei — APPROVED** [review]

File: demos/c_api_minimal_app/Makefile [hunk 1]
```diff
@@ -18,13 +18,13 @@
 HTTP_PROXY := "$(http_proxy)"
 HTTPS_PROXY := "$(https_proxy)"
 NO_PROXY := "$(no_proxy)"
-BASE_OS_TAG_UBUNTU ?= 20.04
+BASE_OS_TAG_UBUNTU ?= 22.04
 BASE_OS_TAG_REDHAT ?= 8.8
-REDHAT_PACKAGE_URL ?="https://github.com/openvinotoolkit/model_server/releases/download/v2023.2/ovms_redhat.tar.gz"
-UBUNTU_PACKAGE_URL ?="https://github.com/openvinotoolkit/model_server/releases/download/v2023.2/ovms_ubuntu20.tar.gz"
+REDHAT_PACKAGE_URL ?="https://github.com/openvinotoolkit/model_server/releases/download/v2023.3/ovms_redhat.tar.gz"
+UBUNTU_PACKAGE_URL ?="https://github.com/openvinotoolkit/model_server/releases/download/v2023.3/ovms_ubuntu22.tar.gz"
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply demos/c_api_minimal_app/Makefile:24]: Packages are ready: https://github.com/openvinotoolkit/model_server/releases/tag/v2024.0

File: demos/c_api_minimal_app/capi_files/Dockerfile.ubuntu [hunk 2]
```diff
@@ -37,10 +37,12 @@ RUN	set -e ; \
         groupadd --gid 5000 ovms ; \
         useradd --home-dir /home/ovms --create-home --uid 5000 --gid 5000 --groups 39,44 --shell /bin/bash --skel /dev/null ovms
 
-ARG BUILD_DEPENDENCIES="build-essential libxml2 curl libpugixml1v5 libtbb2"
+ARG BUILD_DEPENDENCIES="build-essential libxml2 curl libpugixml1v5 libtbb2 wget"
 
 RUN apt-get update && \
     apt-get install -y --no-install-recommends ${BUILD_DEPENDENCIES} && \
+    wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.22_amd64.deb && \
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply demos/c_api_minimal_app/capi_files/Dockerfile.ubuntu:44]: Is this something we should add to required libraries? https://docs.openvino.ai/2024/ovms_docs_deploying_server.html#deploying-model-server-on-baremetal-without-container

[the remark this answers is not in the document]
Comment (bstrzele) [reply demos/c_api_minimal_app/capi_files/Dockerfile.ubuntu:44]: > Why libssl?

it is required for Jenkins job to pass

[the remark this answers is not in the document]
Comment (bstrzele) [reply demos/c_api_minimal_app/capi_files/Dockerfile.ubuntu:44]: > Is this something we should add to required libraries? [docs.openvino.ai/2024/ovms_docs_deploying_server.html#deploying-model-server-on-baremetal-without-container](https://docs.openvino.ai/2024/ovms_docs_deploying_server.html#deploying-model-server-on-baremetal-without-container)

likely, I'll check it out

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**